raiden/waiting.py

```diff
@@ -339,3 +341,34 @@ def wait_for_transfer_success(
                 break
 
         gevent.sleep(retry_timeout)
+
+
+def wait_for_withdraw_complete(
+    raiden: "RaidenService",
+    canonical_identifier: CanonicalIdentifier,
+    total_withdraw: WithdrawAmount,
+    retry_timeout: float,
+) -> None:
+    """Wait until a transfer with a specific identifier and amount
+    is seen in the WAL.
+
+    Note:
+        This does not time out, use gevent.Timeout.
+    """
+    found = False
+    while not found:
+        assert raiden, TRANSPORT_ERROR_MSG
+        assert raiden.wal, TRANSPORT_ERROR_MSG
+        assert raiden.transport, TRANSPORT_ERROR_MSG
+
+        state_changes = raiden.wal.storage.get_state_changes()
```

Comment:
I'd suggest two optimizations:
- use the `offset` parameter on consecutive attempts
- instead of returning deserialized state_changes, `get_state_changes` could be implemented as a generator, that 
```
for state_change in state_changes:
    yield self.serializer.serialize(state_change)
```
this way, breaking below saves us some unnecessary serializations + and we don't need to load everything into memory.

Reply:
We can probably do this in a separate PR if you don't mind

Reply:
sure